interfaces/builtin/desktop.go

```diff
@@ -609,9 +617,51 @@ func (iface *desktopInterface) AppArmorPermanentSlot(spec *apparmor.Specificatio
 	return nil
 }
 
+// https://specifications.freedesktop.org/desktop-entry-spec/latest/file-naming.html
+// Desktop file id must be a valid D-Bus name:
+//  - A sequence of non-empty elements separated by dots
+//  - None of which starts with a digit
+//  - Each of which contains only characters from the set [A-Za-z0-9-_]
+//
+// XXX: dashes "-" are not recommended but supported, should they be removed?
+// https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names
+var desktopFileIDRegexp = regexp.MustCompile("^([A-Za-z-_][\\w-]*)(\\.[A-Za-z-_][\\w-]*)*$")
```

Comment:
Use a raw string instead of escaping the backslashes.

Reply:
Also I see that the specification uses `[A-Za-z0-9-_]`. But in my opinion, it is is better to put the dash when it is not used for a range as the last character of a class. Like `[A-Za-z0-9_-]`. Because without parsing it from left to right, you can know for sure the dash here is is included in the class.

Reply:
Thanks, updated!